File: robot_examples/haskell-servant/rest_api/lib/Model/RobotConfiguration.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

module Model.RobotConfiguration
    ( RobotConfiguration (..)
    ) where

import Data.Aeson
import GHC.Generics
import Test.QuickCheck


data RobotConfiguration = RobotConfiguration
    { name :: String
    , maxHitPoints :: Float
    , maxSpeed :: Float
    , acceleration :: Float
    , decelleration :: Float
    , maxSterlingSpeed :: Float
    , maxScanDistance :: Float
    , maxFireDistance :: Float
    , bulletSpeed :: Float
    , bulletDamage :: Float
    , cannonReloadingTime :: Float
    } deriving (Show, Eq, Generic)

instance FromJSON RobotConfiguration
instance ToJSON RobotConfiguration
instance Arbitrary RobotConfiguration where
    arbitrary = RobotConfiguration <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary

File: client/generated_rest_api/go/swagger/RobotConfiguration.go
package swagger

import (
)

type RobotConfiguration struct {
    Name  string  `json:"name,omitempty"`
    MaxHitPoints  float32  `json:"maxHitPoints,omitempty"`
    MaxSpeed  float32  `json:"maxSpeed,omitempty"`
    Acceleration  float32  `json:"acceleration,omitempty"`
    Decelleration  float32  `json:"decelleration,omitempty"`
    MaxSterlingSpeed  float32  `json:"maxSterlingSpeed,omitempty"`
    MaxScanDistance  float32  `json:"maxScanDistance,omitempty"`
    MaxFireDistance  float32  `json:"maxFireDistance,omitempty"`
    BulletSpeed  float32  `json:"bulletSpeed,omitempty"`
    BulletDamage  float32  `json:"bulletDamage,omitempty"`
    FireReloadingTime  float32  `json:"fireReloadingTime,omitempty"`
    
}
